pymeasure/display/widgets/filename_widget.py

```diff
@@ -0,0 +1,67 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2023 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import logging
+
+from ..Qt import QtCore, QtWidgets
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+class FilenameLineEdit(QtWidgets.QLineEdit):
+    """
+    Widget that allows to choose a filename.
+    A completer is implemented for quick completion of placeholders
+    """
+
+    def __init__(self, parent=None):
+        super().__init__(parent=parent)
+
```

Comment:
The content of this widget does not describe the full filename, does it?
It is just the base, which will be made unique and will be expanded by the file type extension, isn't it?

I'd add a tooltip, which states, that this is the base of the filename.